azure-keyvault/azure/keyvault/secrets/_models.py

```diff
@@ -8,78 +8,107 @@
 from msrest.serialization import Model
 from msrest.paging import Paged
 from .._internal import _parse_vault_id
+from datetime import datetime
+from typing import Any, Mapping
 
-
-class Secret(Model):
-    """A secret consisting of a value, id and its attributes.
+class SecretAttributes(Model):
+    """A secret's id and attributes.
 
     Variables are only populated by the server, and will be ignored when
     sending a request.
 
-    :param str value: The secret value.
     :param str id: The secret id.
     :param str content_type: The content type of the secret.
-    :param attributes: The secret management attributes.
-    :type attributes: ~azure.keyvault.v7_0.models.SecretAttributes
     :param tags: Application specific metadata in the form of key-value pairs.
     :type tags: dict[str, str]
-    :ivar str kid: If this is a secret backing a KV certificate, then this field
+    :ivar str key_id: If this is a secret backing a KV certificate, then this field
      specifies the corresponding key backing the KV certificate.
     :ivar bool managed: True if the secret's lifetime is managed by key vault. If
      this is a secret backing a certificate, then managed will be true.
     """
 
     _validation = {
-        'kid': {'readonly': True},
+        'key_id': {'readonly': True},
         'managed': {'readonly': True},
     }
 
     _attribute_map = {
-        'value': {'key': 'value', 'type': 'str'},
         'id': {'key': 'id', 'type': 'str'},
         'content_type': {'key': 'contentType', 'type': 'str'},
-        'attributes': {'key': 'attributes', 'type': 'SecretAttributes'},
+        '_management_attributes': {'key': 'attributes', 'type': '_SecretManagementAttributes'},
         'tags': {'key': 'tags', 'type': '{str}'},
         'key_id': {'key': 'kid', 'type': 'str'},
         'managed': {'key': 'managed', 'type': 'bool'},
     }
 
+    def __init__(self, **kwargs):
+        # type: (Mapping[str, Any]) -> None
+        super(SecretAttributes, self).__init__(**kwargs)
+        self.id = kwargs.get('id', None)
+        if self.id:
+            self._vault_id = _parse_vault_id(self.id)
+        self.content_type = kwargs.get('content_type', None)
+        self._management_attributes = kwargs.get('_management_attributes', None)
+        self.tags = kwargs.get('tags', None)
+        self.key_id = None
+        self.managed = None
+
+    @property
+    def enabled(self):
+        # type: () -> bool
+        """Gets the Secret's 'enabled' attribute"""
+        return self._management_attributes.enabled
+
+    @property
+    def not_before(self):
+        # type: () -> datetime
+        """Gets the Secret's 'not_before' attribute"""
+        return self._management_attributes.not_before
+
+    @property
+    def expires(self):
+        # type: () -> datetime
+        """Gets the Secret's 'expires' attribute"""
+        return self._management_attributes.expires
+
```

Comment:
enabled, not_before, and expires need should have setters as well

Reply:
Reasoning here is that these models should be considered immutable because no `SecretClient` method takes one of them as a parameter, i.e. changing a property value has no useful effect. `SecretClient` instead exposes everything mutable as method parameters. (The goal there was to save users from figuring out how to construct models.)